app.py

```diff
@@ -209,34 +209,33 @@ def stop_following(follow_id):
     return redirect(f"/users/{g.user.id}/following")
 
 
-@app.route('/users/profile/<int:user_id>', methods=["GET", "POST"])
-def profile(user_id):
+@app.route('/users/profile', methods=["GET", "POST"])
+def profile():
     """Update profile for current user."""
 
     if not g.user:
         flash("Access unauthorized.", "danger")
         return redirect("/")
     
-    user = User.query.get_or_404(user_id)
     form = EditUserForm()
 
     if form.validate_on_submit():
         password = form.password.data
-        test = User.authenticate(user.username, password)
+        test = User.authenticate(g.user.username, password)
```

Comment:
seems strange. Why don't we use form.username.data here? Is there a guarantee that g.user.username is the one matching the password being sent in?

Reply:
Also test isn't a good variable name here. Let's be specific